src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.dao;
+
+import mate.academy.Exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao{
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save product to DB", e);
```

Comment:
you may want to add more information here, like on which movie exactly exception was thrown.
```suggestion
            throw new DataProcessingException("Can't save movie: " + movie, e);
```

Reply:
Done